packages/@aws-cdk/aws-apigateway/lib/integrations/stepfunctions.ts

```diff
@@ -0,0 +1,237 @@
+    this.action = options.action;
+  }
+
+  public bind(method: Method): IntegrationConfig {
+    const bindResult = super.bind(method);
+    const principal = new iam.ServicePrincipal('apigateway.amazonaws.com');
+
+    this.stateMachine.grantExecution(principal, `states:${this.action}`);
+
+    let stateMachineName;
+
+    if (this.stateMachine instanceof sfn.StateMachine) {
+      //if not imported, extract the name from the CFN layer to reach the
+      //literal value if it is given (rather than a token)
+      stateMachineName = (this.stateMachine.node.defaultChild as sfn.CfnStateMachine).stateMachineName;
+    } else {
+      //imported state machine
+      stateMachineName = 'StateMachine-' + (String(this.stateMachine.stack.node.addr));
+    }
+
+    let deploymentToken;
+
+    if (stateMachineName !== undefined && !Token.isUnresolved(stateMachineName)) {
+      deploymentToken = JSON.stringify({ stateMachineName });
+    }
+    return {
+      ...bindResult,
+      deploymentToken,
+    };
+
+  }
+}
+
+function integrationResponse() {
+  const errorResponse = [
+    {
+      /**
+       * Specifies the regular expression (regex) pattern used to choose
+       * an integration response based on the response from the back end.
+       * In this case it will match all '4XX' HTTP Errors
+       */
+      selectionPattern: '4\\d{2}',
+      statusCode: '400',
+      responseTemplates: {
+        'application/json': `{
+            "error": "Bad input!"
+          }`,
+      },
+    },
+    {
+      /**
+       * Match all '5XX' HTTP Errors
+       */
+      selectionPattern: '5\\d{2}',
+      statusCode: '500',
+      responseTemplates: {
+        'application/json': '"error": $input.path(\'$.error\')',
+      },
+    },
+  ];
+
+  const integResponse = [
+    {
+      statusCode: '200',
+      responseTemplates: {
+        'application/json': `#set($inputRoot = $input.path('$'))
+                #if($input.path('$.status').toString().equals("FAILED"))
+                    #set($context.responseOverride.status = 500)
+                    { 
+                      "error": "$input.path('$.error')",
+                      "cause": "$input.path('$.cause')"
+                    }
+                #else
+                    $input.path('$.output')
+                #end`,
+      },
+    },
+    ...errorResponse,
+  ];
+
+  return integResponse;
+}
+
+function requestTemplates(stateMachine: sfn.IStateMachine, includeRequestContext: boolean, requestContextObj: RequestContext | undefined) {
+  const templateStr = templateString(stateMachine, includeRequestContext, requestContextObj);
+
+  const requestTemplate: { [contentType:string] : string } =
+    {
+      'application/json': templateStr,
+    };
+
+  return requestTemplate;
+}
+
+function templateString(stateMachine: sfn.IStateMachine, includeRequestContext: boolean, requestContextObj: RequestContext | undefined): string {
+  let templateStr: string;
+
+  if (includeRequestContext) {
```

Comment:
What if everything was set to `false`? Should this be considered as request context being included or not?